app/models/topic_view_item.rb

```diff
@@ -11,7 +11,8 @@ class TopicViewItem < ActiveRecord::Base
 
   def self.add(topic_id, ip, user_id = nil, at = nil, skip_redis = false)
     # Only store a view once per day per thing per (user || ip)
-    at ||= Date.today
+    at ||= Time.zone.today
```

Comment:
The flaky test in question:
https://github.com/discourse/discourse/blob/44fe8c62d639fb2d202e629de36e069d3c1d46b9/spec/system/request_tracker_spec.rb#L246